lib/spack/spack/environment/environment.py

```diff
@@ -1541,18 +1559,38 @@ def default_view(self):
 
         return self.views[default_view_name]
 
-    def update_default_view(self, viewpath):
-        name = default_view_name
-        if name in self.views and self.default_view.root != viewpath:
-            shutil.rmtree(self.default_view.root)
+    def update_default_view(self, path_or_bool: Union[str, bool]) -> None:
+        view_path = self.view_path_default if path_or_bool is True else path_or_bool
 
-        if viewpath:
-            if name in self.views:
-                self.default_view.root = viewpath
-            else:
-                self.views[name] = ViewDescriptor(self.path, viewpath)
-        else:
-            self.views.pop(name, None)
+        # We don't have the view, and we want to remove it
+        if default_view_name not in self.views and path_or_bool is False:
+            return
+
+        # We have the view, and we want to set it to the same path
+        if default_view_name in self.views and self.default_view.root == view_path:
```

Comment:
I think this check needs to check whether the current default view (in its entirety) is equal to the view descriptor created from just the path. Because if your default view has `select/exclude/projections` in it already, it's a change if you want to update that to just a path.

Reply:
@becker33 What would be the desired behavior? If I have a complex view and:
```console
$ spack env view enable <path>
```
should I change just the path of the complex view or delete and dump a `ViewDescriptor(self.path, view_path)`?

Reply:
I think it should probably be the former, but either is surprising enough that we should document it explicitly.

Reply:
Should be fixed in 91f0267 with docstring and tests